### DISCUSSION TITLE
ConfirmPasswordReset problem

### DISCUSSION THREAD

**Discussion by BashirrMehdixan** [pocketbase/pocketbase#4809] [Q&A] [answered]
Hello, in my react project I want to change the user password with pocketbase, even though both passwords are the same, I get the following error:

`ClientResponseError 400: ClientResponseError
    at pocketbase.js?v=7f836897:889:13
    at async confirmPass (AuthHooks.js:91:13)
    at async handleConfirmPass (Dashboard.jsx:14:9)
    at async react-hook-form.js?v=7f836897:1629:9`

Here are my codes :

`  const confirmPass = async (data) => {
        try {
            console.log(pb.authStore.token)
            await pb.collection('users').confirmPasswordReset(`${pb.authStore.token}`, `${data.password}`, `${data.passwordConfirm}`);
            toast.success("Your password changed");
            return await pb.collection("users").authWithPassword(`${currentUser.email}`, `${data.password}`);
        } catch (e) {
            console.log(e)
            console.log(pb.authStore.token)
            toast.error(e.message)
        }
    }`

Here are  my html codes: 
`<form onSubmit={handleSubmit(confirmPass)}>
                            <div className="account-form">
                                <div className="form-group">
                                    <label htmlFor="newPassword">
                                        New Password
                                    </label>
                                    <input type="password"
                                           className={"form-control"}
                                           id={"newPassword"}
                                           placeholder={"New Password"}
                                           {...register("password")}
                                    />
                                </div>
                                <div className="form-group">
                                    <label htmlFor="confirmNewPassword">
                                        New Password
                                    </label>
                                    <input type="password"
                                           className={"form-control"}
                                           id={"confirmNewPassword"}
                                           placeholder={"Confirm New Password"}
                                           {...register("confirmPassword")}
                                    />
                                </div>
                                <button type="submit" className={"btn btn-blue"}>Update</button>
                            </div>
                        </form>`

**Comment by ganigeorgiev** [comment]
`pb.authStore.token` is an auth token and will not work because `confirmPasswordReset` expects a password reset token. The password reset token can be extracted from the url that the client is redirecting to when clicking on the email link.

**Reply by ganigeorgiev** [reply]
Additionally you don't need to wrap in a template literal the `confirmPasswordReset` and the `authWithPassword` arguments (_unless the values are not a string and you are using it as cast?_).

**Reply by BashirrMehdixan** [reply]
How can extract token from url? And how can I use "confirmPasswordReset"?

**Reply by ganigeorgiev** [reply, marked as answer]
I don't understand what you are asking and I'm starting to think that you are confusing what `confirmPasswordReset()` is intended for.

If you just want to change the password of your currently authenticated user you can send an `update` request:

```js
await pb.collection("users").update(pb.authStore.model.id, {
    oldPassword:     "yourOldPassword",
    password:        "yourNewPassword",
    passwordConfirm: "yourNewPassword",
})
```

`confirmPasswordReset()` is expected to be used as part of the "forgotten password"/"password reset" flow (aka. when the user doesn't know their old password) when you want to specify **your own custom "Password reset confirm" page**.

This means that you have to first specify a custom "Action URL" in the _Admin UI > Settings > Mail settings > Default "Password reset" email template_ that will point to your custom "Password reset confirm" page. The `{TOKEN}` placeholder can be part of your router path or as query parameter (eg. `?token={TOKEN}`). This will depend on your desired behavior and frontend application router.

Or in other words, the flow is usually:

1. Call `requestPasswordReset(email)`
2. The user receives an email with link that contains the token (aka. this the "Action URL").
3. The user clicks on the received email url and it is navigated to your "Password reset confirm" page where they can enter their new password.
4. You have to extract the token from the url (either as router path or query param) and then call `confirmPasswordReset(token, passord, passwordConfirm)`.

I don't know how to explain it more clear than this and will stop engage to avoid further confusion.

**Reply by BashirrMehdixan** [reply]
Thank you so much
